Donation-platform/src/main/java/com/google/servlets/PersonalizationServlet.java

```diff
@@ -48,8 +53,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       e.printStackTrace();
     }
     // Iterate over the mapping and add the tag values to the tags list
```

Comment:
check size of map instead of iterating over

Reply:
Iterating over the map is also needed in order to add tags to the final tags list.

Reply:
You can get all values of a map using map.values()

Reply:
Yeah that's true. But because I have to iterate in order to check for blank tags anyways, I think adding the tag values to the list within the same iteration is slightly more efficient.